MongoDB - Case Study

Jenfer93 opened the pull request:
A case study about MongoDB created by Jenny Feragen. 

* Brief history
* Features
* Strengths 
* Weaknesses 
* Compared to CouchDB (another NoSQL database) 
* Summary 

Considering rewriting the comparison as I talk some about the difference between SQL and NoSQL, should I maybe compare to SQL instead?

Fermain commented on mongoDB-case-study.md:
```diff
@@ -0,0 +1,113 @@
+
+## Introduction to MongoDB
+
+MongoDB is a open source NoSQL database management program that has focus on being scalable and flexible, allowing for storing of big amounts of data in a fast and agile database.
```
```suggestion
MongoDB is a open source NoSQL database management program that has a focus on being scalable and flexible, allowing for the storage of large amounts of data in a fast and agile database.
```

Fermain commented on mongoDB-case-study.md:
```diff
@@ -0,0 +1,113 @@
+
+## Brief History
+
+In the early days, 1970s, NoSQL was just a supplement to the SQL ways of storing data. But as the internet grew and evolved, so did the users, the records and the size of data being stored. The SQL had limitations, and they became more and more apparent. So adopting a more flexible and scalable way to store data was a priority. This formed the evolution of NoSQL, and from that MongoDB arised.
```
```suggestion
In the early days, 1970s, NoSQL was just a supplement to the SQL ways of storing data. But as the internet grew and evolved, so did the users, the records and the size of data being stored. SQL had limitations, and they became more and more apparent. Adopting a more flexible and scalable way to store data was a priority. This began the evolution of NoSQL, and from that MongoDB arose.
```

Fermain commented on mongoDB-case-study.md:
```diff
@@ -0,0 +1,113 @@
+
+In the early days, 1970s, NoSQL was just a supplement to the SQL ways of storing data. But as the internet grew and evolved, so did the users, the records and the size of data being stored. The SQL had limitations, and they became more and more apparent. So adopting a more flexible and scalable way to store data was a priority. This formed the evolution of NoSQL, and from that MongoDB arised.
+
+In the points below, I will list the startup points to MongoDB, but also the biggest releases added to the database throughout the years.
```
In this style of writing we use:
- We `now we can look at this feature...`
- You `you can check this for yourself by visiting...`
- The author `In the opinion of the author, this is not enough...`

But we do not use:
- I

I is only used in opinion writing, whereas this is a technical document.

Fermain commented on mongoDB-case-study.md:
```diff
@@ -0,0 +1,113 @@
+In the points below, I will list the startup points to MongoDB, but also the biggest releases added to the database throughout the years.
+
+- 2007 - MongoDB was founded by Dwight Merriman, Eliot Horrowitz and Kevin Ryan.
+- 2007 - MongoDB the work begins and it was initially developed as a component of a planning platform as a service platform.
```
This sentence should be rephrased as it's difficult to read.

Fermain commented on mongoDB-case-study.md:
```diff
@@ -0,0 +1,113 @@
+## MongoDBs main Features
+
+- Indexing
+  - This makes the searching though the database less time consuming and makes it easier. This means that every field in the documents are indexed with primary and secondary indices.
```
Split this into two bullets

Fermain commented on mongoDB-case-study.md:
```diff
@@ -0,0 +1,113 @@
+- Indexing
+  - This makes the searching though the database less time consuming and makes it easier. This means that every field in the documents are indexed with primary and secondary indices.
+- Replication
+  - This means the database makes copies of the data and sends it to multiple servers. Which make the data more available and redundant, if one server fails, the data can be retrieved from another server. A replica set can contain of two or more copies of the data, each set may act in the role of primary or secondary replica. Where it is the primary replica that serves as the main server and inte ...
```
Split

Fermain commented on mongoDB-case-study.md:
```diff
@@ -0,0 +1,113 @@
+- Replication
+  - This means the database makes copies of the data and sends it to multiple servers. Which make the data more available and redundant, if one server fails, the data can be retrieved from another server. A replica set can contain of two or more copies of the data, each set may act in the role of primary or secondary replica. Where it is the primary replica that serves as the main server and inte ...
+- Aggregation
+  - Makes it possible to perform operations on the grouped data, and then receive a single or computed result. This is thanks to the three different aggregation methods; aggregation pipeline, map-reduce function and a single-purpose aggregation methods.
```
You could also split this up

Fermain commented on mongoDB-case-study.md:
```diff
@@ -0,0 +1,113 @@
+- Programming languages - they have official drives for multiple programming languages, both major and a large number of smaller once.
+
+## Weaknesses
+
```
You should mention ACID here

Jenfer93 replied:
I've fixed the other issues/suggestions, this is a good point and I will need some more time to dig into this!

Fermain commented on mongoDB-case-study.md:
```diff
@@ -0,0 +1,113 @@
+
+## Summary
+
+Overall MongoDB is a strong NoSQL database. The strengths trumps over the weaknesses. MongoDB is easy to use, it has many features many developers look for, and it supports many different development languages, which makes it more accessible for different types of developers. It can contain big amounts of data and the data is easy accessed.
```
```suggestion
Overall MongoDB is a strong NoSQL database. In the author's view, the strengths trump the weaknesses. MongoDB is easy to use, it has many features many developers look for, and it supports many different development languages, which makes it more accessible for different types of developers. It can contain large amounts of data, and the data is easily accessed.
```

Fermain reviewed (changes requested):
This submission is an interesting overview of MongoDB and it's competition. There are some grammar/phrasing edits to be made and I would like to see some discussion of `ACID` as this is arguably the most important weakness of NoSQL.

Perhaps you could compare some situations where SQL or NoSQL would be a good fit? Social media site vs a banking system for example.

joakimlees commented:
Great case-study Jenny! good job.
Left some feedback. Good luck on your delivery!

Jenfer93 commented:
I've added ACID under features, but also added to the ACID part that there are weaknesses to it. I don't think I will include the SQL and NoSQL comparison in this case study, feel like that would deserve a case study for itself. If I start writing about that in this, it will be a very long case-study. Hope that is understandable. @Fermain

Fermain reviewed (approved):
This is a well researched and presented case study on an interesting topic. It serves as a good introduction by highlighting the key information without overwhelming the reader.

The PR shows evidence that you can work with and implement constructive criticism, even on abstract/arcane issues. This is a key skill when writing technical literature.

Well done, and thanks for your contribution!
